### PR TITLE
add cgal

### PR DESCRIPTION
**Pull request by xantares**

### CODE REVIEW

File: recipes/cgal/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  commands:
+    - test -f ${PREFIX}/include/CGAL/AABB_tree.h  # [unix]
+    - test -f ${PREFIX}/lib/libCGAL.so  # [linux]
+    - test -f ${PREFIX}/lib/libCGAL.dylib  # [osx]
+  imports:
+    - CGAL
+
+about:
+  home: http://www.cgal.org/
+  license: GPL/LGPL
```
Comment (jakirkham) [inline recipes/cgal/meta.yaml]: Please specify the license version(s). Is this dual licensed?

File: recipes/cgal/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - cmake
+    - boost
+    - mpfr
+    - zlib
+    - eigen
+    - swig
+  run:
+    - python
+    - boost
+    - mpfr
```
Comment (jakirkham) [inline recipes/cgal/meta.yaml]: Please update these to match our pinning wiki.

File: recipes/cgal/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  version: {{ version }}
+
+source:
+    fn: CGAL-{{ version }}.tar.xz
+    url: https://github.com/CGAL/cgal/releases/download/releases%2FCGAL-{{ version }}/CGAL-{{ version }}.tar.xz
+    md5: 1cbcb3f251822766ed3d744f923c782d
+
+build:
+  number: 1
+# needs GMP on windows
```
Comment (jakirkham) [inline recipes/cgal/meta.yaml]: We have `mpir` on Windows if you want to give that a try. It is basically a cross-platform `gmp`. Though we normally use it on Windows only.

File: recipes/cgal/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  name: cgal
+  version: {{ version }}
+
+source:
+    fn: CGAL-{{ version }}.tar.xz
+    url: https://github.com/CGAL/cgal/releases/download/releases%2FCGAL-{{ version }}/CGAL-{{ version }}.tar.xz
+    md5: 1cbcb3f251822766ed3d744f923c782d
+
+build:
+  number: 1
```
Comment (jakirkham) [inline recipes/cgal/meta.yaml]: Please reset to `0`.

File: recipes/cgal/build.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+site_dir=`python -c "from distutils import sysconfig; print(sysconfig.get_python_lib())"`
+
+cmake \
+  -DCMAKE_FIND_ROOT_PATH=${PREFIX} \
+  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+  -DPYTHON_OUTDIR_PREFIX=${site_dir} \
+  -DCOMMON_LIBRARIES_PATH=${PREFIX}/lib \
+  -DBUILD_JAVA=OFF \
+  ..
+make
```
Comment (jakirkham) [inline recipes/cgal/build.sh]: Is there a `make check`? Also, don't we need `make install`?

File: recipes/cgal/build.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+mkdir -p build && cd build
+
+# needs qt5 for imageio
+cmake \
+  -DCMAKE_FIND_ROOT_PATH=${PREFIX} \
+  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+  -DWITH_CGAL_ImageIO=OFF -DWITH_CGAL_Qt5=OFF \
+  ..
+
```
Comment (jakirkham) [inline recipes/cgal/build.sh]: Is there a `make check`?

File: recipes/cgal/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    - CGAL
+
+about:
+  home: http://www.cgal.org/
+  license: GPL/LGPL
+  summary: Computational Geometry Algorithms Library
+
+extra:
+  recipe-maintainers:
+    - xantares
```
Comment (jakirkham) [inline recipes/cgal/meta.yaml]: nit: please add a terminal newline.

File: recipes/cgal/build.sh [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+make install
+
+
+# language bindings are in a separate repo without releases
+git clone https://github.com/CGAL/cgal-swig-bindings.git
```
Comment (jakirkham) [inline recipes/cgal/build.sh]: I kind of wonder if we shouldn't break this into two packages. One for `cgal` and one for the `python`-bindings. Thoughts @ocefpaf @pelson?

File: recipes/cgal/meta.yaml (hunk 1 above)
Comment (xantares) [inline recipes/cgal/meta.yaml]: As I understand some modules are GPL and some only LGPL

File: recipes/cgal/meta.yaml (hunk 3 above)
Comment (xantares) [inline recipes/cgal/meta.yaml]: Nah I get compilation erros altough it's properly detected

File: recipes/cgal/build.sh (hunk 5 above)
Comment (xantares) [inline recipes/cgal/build.sh]: An install rule is on the way:
https://github.com/CGAL/cgal-swig-bindings/pull/71
For the tests it's not in good shape either, but I proposed a travis script too.

File: recipes/cgal/build.sh (hunk 6 above)
Comment (xantares) [inline recipes/cgal/build.sh]: nope, no make test either

File: recipes/cgal/meta.yaml (hunk 7 above)
Comment (xantares) [inline recipes/cgal/meta.yaml]: ok

File: recipes/cgal/build.sh (hunk 8 above)
Comment (xantares) [inline recipes/cgal/build.sh]: It crossed my mind but I chose to keep it simple.

File: recipes/cgal/build.sh (hunk 6 above)
Comment (xantares) [inline recipes/cgal/build.sh]: more exactly it seems its shipped separately

File: recipes/cgal/build.sh (hunk 8 above)
Comment (pelson) [inline recipes/cgal/build.sh]: Simple is good. If we went ahead with it is as it is, does that make breaking it up later harder?

Comment (xantares) [inline recipes/cgal/build.sh]: So is this ready to go ?

File: recipes/cgal/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+source:
+  fn: CGAL-{{ version }}.zip
+  url: https://github.com/CGAL/cgal/releases/download/releases%2FCGAL-{{ version }}/CGAL-{{ version }}.zip
+  md5: 2780a9c046cbbf26838b8275d2719d5c
+
+build:
+  number: 0
+# fails with mpir as replacement of gmp on windows
+  skip: true  # [win]
```
Comment (patricksnape) [inline recipes/cgal/meta.yaml]: So Windows is not working? Can we remove any windows logic (`bld.bat` etc) and then add them as a PR on the feedstock to keep the recipe clean?

Comment (xantares) [inline recipes/cgal/meta.yaml]: @patricksnape, ok done

File: recipes/cgal/build.sh [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+cd cgal-swig-bindings
+mkdir -p build && cd build
+site_dir=`python -c "from distutils import sysconfig; print(sysconfig.get_python_lib())"`
```
Comment (patricksnape) [inline recipes/cgal/build.sh]: Can use the environment variable provided by conda: `SP_DIR`

Comment (xantares) [inline recipes/cgal/build.sh]: @patricksnape, ok using it now, thanks

**Comment by patricksnape** [comment]
@jakirkham I think this looks good to me now.

File: recipes/cgal/build.sh (hunk 8 above)
Comment (jakirkham) [inline recipes/cgal/build.sh]: I'm not sure that answers @pelson's question.

File: recipes/cgal/meta.yaml (hunk 7 above)
Comment (jakirkham) [inline recipes/cgal/meta.yaml]: Done.

**Comment by jakirkham** [comment]
I think it looks alright too. Though @pelson asked a great question IMHO and I would love to hear the answer.

File: recipes/cgal/build.sh (hunk 8 above)
Comment (xantares) [inline recipes/cgal/build.sh]: @jakirkham, yeah well I guess we could eventually split it by adding a cgal-libs package. The cgal package would continue providing the python bindings as expected.

**Comment by xantares** [comment]
@jakirkham, I guess we could eventually split it by adding a cgal-libs package for the lib only. The cgal package would continue providing the python bindings as expected.

**Comment by xantares** [comment]
so ?

**Comment by patricksnape** [comment]
@xantares Do the swig bindings link dynamically to cgal or do they recompile the source? Because if they just link dynamically then we may as well split it now. Otherwise, splitting it would be basically impossible anyway.

**Comment by xantares** [comment]
@patricksnape the swig bindings link dynamically so it's feasible to split. But again, should we ? Unless I hear arguments in favor of the swigxit :) I would prefer not to split it : "keep it simple", remember ?
Back to you guys.

**Comment by patricksnape** [comment]
Honestly I'm not dogmatic, if there are other bindings then it makes sense, merely because it might be confusing down the line to suddenly have cgal no longer installing Python bindings when all the bindings become split. In the interest of not holding everyone up though, I'd be fine to leave as is (especially since i'm not a cgal user 😛).

**Comment by xantares** [comment]
@patricksnape There are no other python bindings available, so I guess this settles it.

**Comment by jakirkham** [comment]
So you, @xantares, should get an email that invite you to join conda-forge. Once accepted you will be added to a team with the same name as this recipe. Those will give you permissions on the feedstock (repo) for this recipe. Make sure when proposing any change that you go through the typical GitHub workflow of forking the feedstock and making changes in your fork that you PR back. Once merged CIs will build and deploy any changes you make. Please let us know if you have any questions and welcome to conda-forge.

**Pull request merged**